packages/renderer/src/lib/feedback/feedbackForms/DevelopersFeedback.spec.ts

```diff
@@ -0,0 +1,184 @@
+/**********************************************************************
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import '@testing-library/jest-dom/vitest';
+
+import { fireEvent, render, screen } from '@testing-library/svelte';
+import userEvent from '@testing-library/user-event';
+import { beforeEach, expect, test, vi } from 'vitest';
+
+import DevelopersFeedback from './DevelopersFeedback.svelte';
+
+const openExternalMock = vi.fn();
+const telemetryTrackMock = vi.fn();
+const sendFeedbackMock = vi.fn();
+
+beforeEach(() => {
+  vi.resetAllMocks();
+  Object.defineProperty(window, 'openExternal', {
+    value: openExternalMock,
+  });
+  Object.defineProperty(window, 'telemetryTrack', {
+    value: telemetryTrackMock,
+  });
+  Object.defineProperty(window, 'sendFeedback', {
+    value: sendFeedbackMock,
+  });
```

Comment:
what is the issue with the single defineProperty ?

AFAIK you should put it in beforeAll block and not beforeEach to make it work properly (to avoid issue about trying to redefine something)

Reply:
I'm not sure why, but for some reason, using the Object.defineProperty(global, 'window' pattern (in a beforeAll) caused the tests to fail (rendered components from previous tests "leaked" into the tests after them), so I used Object.defineProperty(window, '<method name>' for each one

Reply:
ok, it's not blocking the PR anyway